### PR TITLE
[tbl111] Use pipenv for deterministic builds

### PR DESCRIPTION
**Pull request by jcox-dev**
Pipenv enables us to lock down and guarantee what dependencies we get
when we install dependencies. This works by using the pip hashes when
running pipenv install --dev. The hashes are stored in a generated
file call Pipfile.lock.

We no longer need requirement files instead pipenv uses a file
called Pipfile. The documentation can be found
https://github.com/kennethreitz/pipenv.

What is the context of this PR?

Link to Trello card

https://trello.com/c/TNrTo9L1/733-tbl111-create-reproducible-deployments

How to review

Check out locally and follow README.md to set up with pipenv
cf push to check deployment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,8 +59,8 @@ to supply valid UUID's when testing.
 For testing, you can upload from the **scripts** folder, you should be able to run;
 
 ```bash
-ce=`python3 -c "from uuid import uuid4; print(uuid4())"`
-./upload.py local $ce 30
+ce=`pipenv run python -c "from uuid import uuid4; print(uuid4())"`
+pipenv run ./upload.py local $ce 30
```
Comment (benjefferies) [inline README.md]: Looks like the port in this script is wrong. It points to 8080 but the app runs up on 8082 and then I get this error

```
%% upload error "404" - "{
  "detail": "The requested URL was not found on the server.  If you entered the URL manually please check your spelling and try again.",
  "status": 404,
  "title": "Not Found",
  "type": "about:blank"
}
"
```

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+
+[requires]
+
+python_version = "3.6.3"
+
+
+[packages]
+
+ras-common = {git = "https://github.com/ONSdigital/ras-common.git", ref="0.3.0"}
```
Comment (benjefferies) [inline Pipfile]: Good job at cutting this down

File: tox.ini [hunk 3]
```diff
@@ -3,13 +3,13 @@ envlist = py36
```
Comment (benjefferies) [inline tox.ini]: Getting the following running `pipenv run tox`
```
Traceback (most recent call last):
  File "/Users/benjefferies/.virtualenvs/ras-collection-instrument-QrNRiIIx/bin/tox", line 11, in <module>
    sys.exit(cmdline())
  File "/Users/benjefferies/.virtualenvs/ras-collection-instrument-QrNRiIIx/lib/python3.6/site-packages/tox/session.py", line 39, in main
    config = prepare(args)
  File "/Users/benjefferies/.virtualenvs/ras-collection-instrument-QrNRiIIx/lib/python3.6/site-packages/tox/session.py", line 27, in prepare
    config = parseconfig(args)
  File "/Users/benjefferies/.virtualenvs/ras-collection-instrument-QrNRiIIx/lib/python3.6/site-packages/tox/config.py", line 245, in parseconfig
    parseini(config, inipath)
  File "/Users/benjefferies/.virtualenvs/ras-collection-instrument-QrNRiIIx/lib/python3.6/site-packages/tox/config.py", line 810, in __init__
    config.envconfigs[name] = self.make_envconfig(name, section, reader._subs, config)
  File "/Users/benjefferies/.virtualenvs/ras-collection-instrument-QrNRiIIx/lib/python3.6/site-packages/tox/config.py", line 838, in make_envconfig
    res = meth(env_attr.name, env_attr.default, replace=replace)
  File "/Users/benjefferies/.virtualenvs/ras-collection-instrument-QrNRiIIx/lib/python3.6/site-packages/tox/config.py", line 1011, in getargvlist
    return _ArgvlistReader.getargvlist(self, s, replace=replace)
  File "/Users/benjefferies/.virtualenvs/ras-collection-instrument-QrNRiIIx/lib/python3.6/site-packages/tox/config.py", line 1199, in getargvlist
    (reader.section_name, "commands"))
tox.ConfigError: ConfigError: line-continuation ends nowhere while resolving for [testenv:py36] commands
```

File: README.md (hunk 1 above)
Comment (benjefferies) [reply README.md]: Probably not worth looking into it on this PR

File: tox.ini (hunk 3 above)
Comment (jcox-dev) [reply tox.ini]: Snipped off the [] for some reason.

**Comment by jcox-dev** [comment]
Will probably need some amendments with changes from #50

**Pull request merged**